url.bs

```diff
@@ -756,14 +756,14 @@ not equivalent and typically treated as distinct.
 
 <p>An <dfn export>IP address</dfn> is an <a>IPv4 address</a> or an <a>IPv6 address</a>.
 
-<p>An <dfn export id=concept-ipv4>IPv4 address</dfn> is a 32-bit unsigned integer that identifies a
-network address.
+<p>An <dfn export id=concept-ipv4>IPv4 address</dfn> is a <a for=/>32-bit unsigned integer</a> that
+identifies a network address.
 [[RFC791]]
 
-<p>An <dfn export id=concept-ipv6>IPv6 address</dfn> is a 128-bit unsigned integer that identifies a
-network address. For the purposes of this standard it is represented as a <a for=/>list</a> of eight
-16-bit unsigned integers, also known as
-<dfn export lt="IPv6 piece" id=concept-ipv6-piece>IPv6 pieces</dfn>.
+<p>An <dfn export id=concept-ipv6>IPv6 address</dfn> is a <a for=/>128-bit unsigned integer</a> that
+identifies a network address. This integer is composed of a <a for=/>list</a> of 8
+<a for=/>16-bit unsigned integers</a>, also known as an <a for=/>IPv6 address</a>'s
+<dfn export for="IPv6 address" id=concept-ipv6-piece>pieces</dfn>.
```

Comment:
This definition is obscure: it is not clear how to convert a 128-bit integer to a list of pieces and vice versa. Also, all algorithms use an IPv6 address as a list of pieces, not as a 128-bit integer. By the way, RFC4291 defines an IPv6 address slightly differently: "IPv6 addresses are 128-bit identifiers...". So I think it is better to use the RFC4291 definition - replace "unsigned integer" with "identifier":
> An IPv6 address is a 128-bit **identifier** that identifies a network address. This **identifier** is composed of ...

Reply:
All algorithms inside URL, yes, but outside? This we would have to change separately though as changing it from integer to identifier would not be an editorial change.

Reply:
I do agree that this is a bit confusing. I can see how it's "conceptually" a u128, or "corresponds to" a u128, but in practice every part of the spec treats it as a list of 8 u16s.